src/meshapi/views/forms.py

```diff
@@ -134,6 +138,50 @@ def join_form(request: Request) -> Response:
             {"detail": "Your address could not be validated."}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
         )
 
+    changed_info: dict[str, str | int] = {}
+
+    # TODO: Notify member if we changed any of their information
+    # Name (won't touch), email (won't touch), phone, st addr, unit (won't touch), city, State, Zip
+    if formatted_phone_number and r.phone != formatted_phone_number:
+        logging.warning(f"Changed phone_number: {formatted_phone_number} != {r.phone}")
+        changed_info["phone"] = formatted_phone_number
+
+    if r.street_address != nyc_addr_info.street_address:
+        logging.warning(f"Changed street_address: {r.street_address} != {nyc_addr_info.street_address}")
+        changed_info["street_address"] = nyc_addr_info.street_address
+
+    if r.city != nyc_addr_info.city:
```

Comment:
Can the ZIP or state mismatch? Or are we excluding that possibility already above?

Reply:
They totally can, but I think I have a test for it now, (at least with NJ state but NYC address). I think it's acceptable to have this validation in the backend with the rest of our address code, but maybe we should have some explicit check for it so we can return a more specific error. Right now, it just tells the member that Non-NYC registrations are not accepted.